### PR TITLE
Polymer 2 header migration 

### PR DESCRIPTION
**Pull request by daniel-tabarcea**
- connects unicef/etools-issues#1150

### CODE REVIEW

File: pmp/modules/layout/components/data-refresh-dialog.html [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+      </div>
+    </etools-dialog>
+
+  </template>
+
+  <script>
+    /**
+     * @polymer
+     * @mixinFunction
+     * @appliedMixin EtoolsLogsMixin
```
Comment (adi130987) [inline pmp/modules/layout/components/data-refresh-dialog.html]: `@appliesMixin`

File: pmp/modules/layout/components/data-refresh-dialog.html [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+     * @appliedMixin EtoolsPageRefreshMixin
+    */
+    const DataRefreshDialogMixin = EtoolsMixinFactory.combineMixins(
+        [EtoolsLogsMixin, EtoolsPageRefreshMixin],
+        Polymer.Element);
+    /**
+     * @polymer
+     * @customElement
+     * @appliedMixin PageHeaderMixin
```
Comment (adi130987) [inline pmp/modules/layout/components/data-refresh-dialog.html]: `@appliesMixin`

File: pmp/modules/layout/components/data-refresh-dialog.html [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+          page: {
+            type: String,
+            notify: true
+          }
+        };
+      }
+
+      static get observers() {
+        return [
+          '_singleSectionChanged(interventionsSelected, partnersSelected, agreementsSelected)',
```
Comment (adi130987) [inline pmp/modules/layout/components/data-refresh-dialog.html]: in Polymer 2 observers run even if arguments are undefined... you might need to add some validation for this

File: pmp/modules/layout/components/countries-dropdown.html [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        if (e.detail.item.countryId !== this.currentCountry.id) {
+          //send post request to change_coutry endpoint
+          let selectedCountryId = e.detail.item.countryId;
+          this._triggerCountryChangeRequest(selectedCountryId);
+        }
+      }
+
+      _triggerCountryChangeRequest(countryId) {
+        this.dispatchEvent(new CustomEvent('global-loading', {
+          message: 'Please wait while country is changing...',
```
[the remark this answers is not in the document]
Comment (adi130987) [reply pmp/modules/layout/components/countries-dropdown.html]: https://www.polymer-project.org/2.0/docs/devguide/events#custom-events

File: pmp/modules/layout/components/data-refresh-dialog.html [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+     * @appliesMixin EtoolsPageRefreshMixin
+     * @appliesMixin EtoolsPmpApp.Mixins.EventHelper
+    */
+    const DataRefreshDialogMixin = EtoolsMixinFactory.combineMixins(
+        [EtoolsLogsMixin, EtoolsPageRefreshMixin, EtoolsPmpApp.Mixins.EventHelper],
+        Polymer.Element);
+    /**
+     * @polymer
+     * @customElement
+     * @appliesMixin PageHeaderMixin
```
Comment (acory) [inline pmp/modules/layout/components/data-refresh-dialog.html]: update @appliesMixin PageHeaderMixin

File: pmp/modules/mixins/event-helper-mixin.html [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<script>
+  'use strict';
+  window.EtoolsPmpApp = window.EtoolsPmpApp || {};
+  window.EtoolsPmpApp.Mixins = window.EtoolsPmpApp.Mixins || {};
+   /*
+  * @polymer
+  * @mixinFunction
+  */
+  EtoolsPmpApp.Mixins.EventHelper = (baseClass) => class extends baseClass {
+
```
Comment (acory) [inline pmp/modules/mixins/event-helper-mixin.html]: very likely that deduppingMixin might be needed here

**Pull request merged**